torch/optim/sgd.py

```diff
@@ -201,19 +218,28 @@ def sgd(params: List[Tensor],
     See :class:`~torch.optim.SGD` for details.
     """
 
-    if foreach is None:
+    if foreach is None and fused is None:
         # why must we be explicit about an if statement for torch.jit.is_scripting here?
         # because JIT can't handle Optionals nor fancy conditionals when scripting
         if not torch.jit.is_scripting():
-            _, foreach = _default_to_fused_or_foreach(params, differentiable=False, use_fused=False)
+            fused, foreach = _default_to_fused_or_foreach(params, differentiable=False, use_fused=False)
```

Comment:
Could you add a comment here similar to in adam(w):
Note that we default to foreach and pass False to use_fused. This is not a mistake--we want to give the fused implementation bake-in time before making it the default, even if it is typically faster.